### DISCUSSION TITLE
H3 load power is incorrect

### DISCUSSION THREAD

**Discussion by benbeton** [nathanmarlor/foxess_modbus#362] [Support]
currently there is no sensors which represents the current energy usage in W (regardless of whether the energy is drawn from the grid, the PV system or the battery. ) for the Fox H3 Inverters

tried this one for "house usage", but there seems to be missing something: should be equal to _sensor.foxess_load_power_ (FoxESS Cloud integration) but is not (close but not complete)

```
- name: "fox house energy current"
        unique_id: fox_house_energy_current
        unit_of_measurement: 'kW'
        device_class: power
        state_class: measurement
        state: >
          {{ ((states('sensor.fox_grid_consumption') | float(default=0)
          + states('sensor.fox_pv_power') | float(default=0)
          + states('sensor.fox_battery_discharge') | float(default=0) ))
          - states('sensor.fox_battery_charge') | float(default=0)  | round(2) }}
```

following can be combined with a template sensor: 
**Grid usage**
```
- name: "fox Grid Consumption"  #positive or zero
        device_class: "power"
        unit_of_measurement: "W"
        state_class: measurement
        state: >
               {{ (states('sensor.fox_grid_consumption_r') | float(default=0)
               + states('sensor.fox_grid_consumption_s') | float(default=0)
               + states('sensor.fox_grid_consumption_t') | float(default=0)) | round(2) }}
```

**Feed In**
```
- name: "fox Feed In Power"  #positive or zero
        device_class: "power"
        unit_of_measurement: "W"
        state_class: measurement
        state: >
               {{ (states('sensor.fox_feed_in_r') | float(default=0)
               + states('sensor.fox_feed_in_s') | float(default=0)
               + states('sensor.fox_feed_in_t') | float(default=0)) | round(2) }}
```
**Load Power**
```
- name: "fox Load Power"
        device_class: "power"
        unit_of_measurement: "W"
        state_class: measurement
        state: >
               {{ (states('sensor.fox_load_power_r') | float(default=0)
               + states('sensor.fox_load_power_s') | float(default=0)
               + states('sensor.fox_load_power_t') | float(default=0)) | round(2) }}
```

would be great if we complete the sensors together

**Comment by canton7** [comment]
There seem to be a number of things going on here.

> currently there is no sensors which represents the current energy usage in W (regardless of whether the energy is drawn from the grid, the PV system or the battery. )

Yes there are. Energy drawn from the grid is `grid_consumption_R/S/T`, energy from the PV system is `pv_power_now`, energy to/from the battery is `battery_charge` / `battery_discharge`

The power taken by your house is *load power*. This is exposed by this integration `load_power_R/S/T`.

> but is not (close but not complete)

This missing piece is that the inverter is not 100% efficient. The difference between your sum and `load_power` will be the inverter losses.

**Reply by benbeton** [reply]
thank you @canton7  maybe I expressed myself wrong, there is not one sensor that displays the current consumption of the house. individual sensors do that as e.g for the consumption from the grid or from the battery. 

I also thought that *.load sensors should show the load of the house, but on my side  its not, maybe there is a bug here!? 
the sensors shows that:

<img width="318" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/0575b313-4393-4b29-8b76-04aea832c417">

 although the current load is:
<img width="541" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/654422ad-bf43-470a-88e2-67aee440093e">

the load sensors showing strange values:
<img width="555" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/0b18248d-c99a-4323-b2a0-5ec61fab3f18">
<img width="554" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/87a951f8-003e-4c64-ae93-85d03db236e2">
<img width="555" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/f44e19a0-c2c3-432b-9565-27ec2e3f4f18">

against it, load from FoxESS Cloud: (which is correct)
<img width="554" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/f1550542-cec0-4bd5-abd8-cd5e80a3fd83">

if, *.load sensors has the correct values, then they can be combined, any ideas!?

**Reply by canton7** [reply]
Interesting, yes that doesn't look right.

On a hunch, can you take a look at Grid CT R/S/T? I wonder whether those two have got mixed up.

@TonyM1958 FYI

**Comment by benbeton** [comment]
Grid CT is showing values, but combined these are not near the correct value: 

**Grid CT**
<img width="511" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/49cc87e2-287f-47ac-9279-5cbda7e34b32">

**Load FoxESS Cloud**
<img width="549" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/426ed62e-886f-4749-8a1a-246a7e762c4b">

**Inverter Power** seems to be near, but also not correct and I think that's not the correct value:
<img width="329" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/8ca46d9a-e9e2-4e68-9ad6-939b236a7900">

**Reply by mafischerhei** [reply]
Ben, can you have a look at my recent post. I think we have a similar "problem":

I think there's a miscalculation in the code for some entities which prevents us from calculating the correct house load


please have a look -> https://github.com/nathanmarlor/foxess_modbus/discussions/456
Thanks

**Reply by benbeton** [reply]
thanks @mafischerhei looks promising - i'll follow you thread!

**Comment by TonyM1958** [comment]
Do you just have 1 x H3 inverter or do you have another solar inverter connected as well? A second inverter will throw out the sensor values on the H3 so its best to understand this first.

**Comment by benbeton** [comment]
it's just 1 H3 Inverter

**Reply by TonyM1958** [reply]
Thanks, connected to a Chint meter sitting in line with you grid supply, or is it connected via current transformers?

**Comment by benbeton** [comment]
there is a Chint DTSU666 sitting in line

**Comment by TonyM1958** [comment]
OK, just to confirm, you are getting the correct R S T power values for your grid consumption and feedin to/from the grid?

From what I know of the sensors on the H1 inverter, Fox may not actually use the inverter load power sensors when the data gets to the cloud because they are not correct in all circumstances. For example, when you add a second inverter, the load power sensor on the inverter returns a -ve value but the Fox cloud returns a +ve value.

For H1, the load power calculation in the cloud may be (Output Power - Grid CT + CT2), where Grid CT is positive for export and -ve for import. This is the sum of the power going through the henley block, so there are no conversion losses (but sampling errors are possible).

The same may apply to H3...

Here's an example of some code I wrote to derrive load power and compare against the inverter value for H1:

```
    - name: "Load Power X"
      device_class: "power"
      unique_id: foxess_inv1_load_power_x
      unit_of_measurement: "kW"
      state: >
        {% set rpower = states('sensor.rpower') | float(default=0) %}
        {% set spower = states('sensor.spower') | float(default=0) %}
        {% set tpower = states('sensor.tpower') | float(default=0) %}
        {% set ct2 = states('sensor.ct2') | float(default=0) %}
        {% set grid = states('sensor.grid_ct') | float(default=0) %}
        {% set rgrid = states('sensor.meter_rpower') | float(default=0) %}
        {% set sgrid = states('sensor.meter_spower') | float(default=0) %}
        {% set tgrid = states('sensor.meter_tpower') | float(default=0) %}
        {% set result = rpower + spower + tpower - grid - rgrid - sgrid - tgrid + ct2 %}
        {{ result }}
```

And an example of how this tracks the inverter load sensor for the H1 base case:

![image](https://github.com/nathanmarlor/foxess_modbus/assets/63789168/4a771e2e-32f1-4c75-b6af-f0a234486482)


You can try to calculate Load Power RST separately as well, but I don't think it will be correct if a second inverter is also connected as you don't have Meter2 Power RST values.

**Comment by TonyM1958** [comment]
BTW - no idea what this means in terms of accuracy for Load Energy Today and Load Energy Total counters on the inverter. I don't think anyone has yet done long term tracking of these counters for more complex cases...

You can do a similar derrived load energy calculation based on Output Energy, Input Energy, Grid Consumption and Feedin Energy but you don't have an energy counter for CT2 / Meter 2. You could try to get generation energy directly from the second inverter, and that total will only be correct if the inverters were installed at the same time?

**Reply by benbeton** [reply]
I do only have one inverter, there has to be a register for the correct load.
for my understanding, this integration has a good register mapping for the H3 inverters, from what aI can see is for example:

**31029** should be "Load Power R" but in this repo, its not tagged  with H3 
<img width="668" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/abb2e812-7d08-4baf-8697-aa8eee2458a4">

**31030** should be "Load Power S" and is tagged here with H3:
<img width="618" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/45defbe5-c82a-4181-9fd8-056c417dd5ac">

source: https://github.com/nathanmarlor/foxess_modbus/blob/300dacf34d087f69288706c614ab9d1e592fa5f0/custom_components/foxess_modbus/entities/entity_descriptions.py#L38

could that be a cause?

**Reply by TonyM1958** [reply]
The registers are listed here: https://github.com/TonyM1958/HA-FoxESS-Modbus/wiki/Modbus-Map.xls

These include Load Power RST as 31029, 31030 and 31031. I thought you had already  looked at these and concluded the values were incorrect?

If you think the registers are not mapped correctly in the integration then this is something @canton7 can investigate.

These still need to taken into account when calculating overall Load Power and the effects on this from CT2 / Meter2 Power injection.

**Reply by canton7** [reply]
I think you needed to click "find next" one more time?

https://github.com/nathanmarlor/foxess_modbus/blob/300dacf34d087f69288706c614ab9d1e592fa5f0/custom_components/foxess_modbus/entities/entity_descriptions.py#L803-L814

**Comment by TonyM1958** [comment]
Been running the calculated load power for a day now to compare with the base case sensor:

![image](https://github.com/nathanmarlor/foxess_modbus/assets/63789168/3f83fb8b-22d9-443a-ab9b-112a0c4d5a2a)

Load Power CT1 is the raw H1 load sensor. Load Power is the value calculated as follows:

![image](https://github.com/nathanmarlor/foxess_modbus/assets/63789168/d134b9d7-6821-41ed-b72e-fafb858ec799)

There's s small adaptation here to stop the value going negative (due to sampling error) as the house load cannot generate. 300w was chosen arbitrarily as an approximation to an average house base load.

You can see around 10:15, the load power spikes up to 5kw but the calculated value spikes to 7kw due to sampling error. I added a filter, load_power_smooth, using a 30 second rolling average to see what it looks like if you smooth out spikes. The filter introduces a small lag in the data transients:

<code>
- name: load_power_smooth
  platform: statistics
  unique_id: foxess_inv1_load_power_smooth
  entity_id: sensor.load_power
  state_characteristic: average_linear
  max_age:
    seconds: 30
  precision: 3
</code>
<br>

Combining the plots highlights the small differences...

![image](https://github.com/nathanmarlor/foxess_modbus/assets/63789168/765a3901-e51b-41f8-8824-ed9d599866cb)


A few options to consider, but it looks like calcuating load power to cover different configurations will work fine where the inverter sensors are not returning the correct values.

**Comment by TonyM1958** [comment]
Wondering if you might have a naming conflict in your system from some previous integration. Have you checked Settings, Devices & Services. Entities and searched for entities with Load in the name and looked for any names with _2 at the end?

**Reply by canton7** [reply]
I'm curious how you think a naming conflict could cause this?

**Reply by TonyM1958** [reply]
Probably a long shot, but if I've read the conversation correctly, you think Load Power RST are probably configured in the integration, but the OP thinks the values are wrong. I've seen many cases where older code has created sensors and when a new integration is added the names unique id or integration name conflict so HA renames the new sensor with _2 at the end.  You can end up looking at the old sensor instead of the new one. Just one thing to rule out but it might explain why this instance might not work when others do?

Other suggestions as to what might be going on or how to collect more data to diagnose welcome...

**Comment by canton7** [comment]
Looks like the Chint might be the culprit. Try restarting it. https://github.com/nathanmarlor/foxess_modbus/issues/371#issuecomment-1645257370

**Comment by benbeton** [comment]
I'll tried, cut off the power of the chint, still the same - but could a configuration mistake of the chint cause the sensor not to show the values?!

**Reply by vip-klaus** [reply]
Hello everyone,

I (a beginner) am dealing with the same problem of calculating current consumption. In my case I use two H3 inverters (master, slave), with only one battery connected to the master.
I noticed that the entities "inv_power_r/s/t" never take negative values, although according to the FoxESS app this can certainly happen while the battery is charging. Namely whenever the master charges the battery with higher power than the PV power (of the master) is available. The negative power then comes from the slave delivering additional AC power to the master.
Can anyone confirm this behavior? Why do the values for "inv_power_R/s/t" never become negative?

Thanks and Greetings
Klaus

**Reply by FozzieUK** [reply]
Hi, the way I was told it should work when connected as master / slave when sharing a common meter (on the master) over canbus, is that you would not expect to see inv_power_r/s/t go negative unless there was not enough PV being generated by both systems **and** you were force charging the batteries from the grid.

The shared meter would allow the inverters to calculate and load balance between themselves, so a negative load situation could only happen when the grid was supplying some of the demand when charging the batteries.

I don't have a lot of experience on the H3's, that is how I was told it would work by Foxess - hopefully someone with a similar system can confirm or otherwise the operation.

**Reply by vip-klaus** [reply]
Hello FozzieUK, thank you for your reply. I don't think the smart meter can detect the power flow between master and slave. It only measures the power between the grid and the house.

Check out the photo of my master inverter from this afternoon:
- Output Power: -3.11kW
- Charge Power: 11.23 kW
- GridConsumption Power: 0 kW

The difference of 2 kW comes from the AC power of the slave. Both the FoxESS smartphone app and the web browser app correctly display the inverter values as negative, but the FoxESS-Modbus integration only recognizes positive values.

Does anyone know which file contains the definition of the entity "inv_power_r/s/t? Maybe an unsigned variable was accidentally defined here? Or I'm missing something else...

Best regards
Klaus

![grid_output_load-Power_22 09 2023](https://github.com/nathanmarlor/foxess_modbus/assets/144471901/54933323-3ee4-44de-a1e5-20a296333b20)

**Reply by FozzieUK** [reply]
Ahhh right, now I understand the problem you've got, - yes you are correct there is a range check on the inv_power_r/s/t and inv_current_r/s/t that doesn't allow these values to be negative.

It'll need @canton7 to have a look at that, but for now you could test it manually by changing the range checks (around lines 621/633/645 of /config/custom_components/foxess_modbus/entities/entity_descriptions.py and make the range check 'validate=[Range(-100, 100)]' and then restart your home assistant.

**Reply by canton7** [reply]
Opened #422

**Reply by vip-klaus** [reply]
Hello FozzieUK, canton7: thank you very much for the super fast help! This was exactly what I was looking for.

Best regards
Klaus

**Reply by canton7** [reply]
Oops... Thanks for spotting and diagnosing this everyone!

I'll get a fix in when I get the chance...

**Reply by vip-klaus** [reply]
Good morning,

I made the changes in the entity_descriptions.py: 'validate=[Range(-100, 100)]' and restarted HA. Unfortunately, the values for inv_power_r/s/t and inv_current_r/s/t are still not represented negatively. And the calculated house consumption is still too high.

There must be an entry somewhere else?

Best regards
Klaus

![Inv_Power_23 09 2023](https://github.com/nathanmarlor/foxess_modbus/assets/144471901/77bfb25a-6208-4c28-b31a-f8fee9583b53)
![HA-Gauges_23 09 2023](https://github.com/nathanmarlor/foxess_modbus/assets/144471901/782900fb-5bfb-4d09-96fd-329a57e09246)

**Reply by TonyM1958** [reply]
Probably need to resolve the sign of RST first but feel free to have a look at this as well: https://github.com/TonyM1958/HA-FoxESS-Modbus/blob/main/custom_components/HA-FoxESS-Modbus/template.yaml#L376

This has been tested on a single H1 and H3 with an external solar inverter and produces the correct load power. That doesn't mean it will be correct where you have an H3 master and 1 or more slaves but it's a starting point for comparison. If the master does not represent the overall RST Power, it's difficult to see you could resolve this without connecting modbus to all of the H3 inverters and reading and combining the power from all the inverters.

**Reply by FozzieUK** [reply]
Thanks Tony, i'm sure it assumes signed as default, but yes duh... I realised with a closer look there is a post process check before the validate which doesn't allow values below 0 in the inv_power_x lines.

@vip-klaus can you remove the post process lines 
delete this line ->   post_process=lambda x: max(x, 0),

In all the locations where you changed the range of inv_power_r/s/t to -100  (lines 620/632/644)

The inv_current_r/s/t are ok - the post process check is not there.

And if you could restart HA and let us know how that goes.

**Reply by vip-klaus** [reply]
@fozzieUK thanks for the tip. I deleted the lines in the file and restarted HA. I won't be able to say until tomorrow whether it works, as the problem only occurs when the battery is massively charged.

But there must also be a problem with the inverter_current_r/s/t: On the screenshot you can see that inverter_current is around 1.4A, even though the inverter_power is "0". If the actual inverter_power is actually negative, then the inverter_current should also be negative, right?

Greetings
Klaus

**Reply by FozzieUK** [reply]
ok, thanks - on the inverter current, with a single phase inverter the current is an absolute value (positive), I think the 3 phase is the same, hopefully that will be confirmed tomorrow when you can see the correct power readings.

**Reply by vip-klaus** [reply]
In the FoxESS app, the inverter_current values (RCurrent, SCurrent, TCurrent) can also become negative. This always happens when the slave inverter delivers AC power to the master to charge the battery to the maximum.

I can give feedback tomorrow, but then I'll be on vacation for a week - without access to my computer.

More will come from October 1st.

Thank you again for your excellent support!

Greetings
Klaus

**Reply by TonyM1958** [reply]
What happens doesn't always work as you might expect as these are AC rather than DC values. Here's what you see on H1 with RPower and RCurrent when charging:

![image](https://github.com/nathanmarlor/foxess_modbus/assets/63789168/ce8c98af-af41-4568-bf78-443c7ce40e9c)

As you can see, RPower goes -ve when charging but RCurrent stays +ve. RVolt is positive for both charging and discharging - so you really need to follow the power values and not multiply volts and current. There are other registers that tell you stuff like real and apparent power - but mostly, people just go by the real power.

**Reply by canton7** [reply]
Let me know what you find -- I've got the change ready to go

**Comment by halfprice11** [comment]
hi folks,

I do face a similar issue - as I try to get rid of the FoxCloud, by building up a local view within home assistand.
Gid, Battery, Solar values are all fine.
Inverter Power does show the values, inverter power output is.

Tried to build something like this "[flixlix](https://github.com/flixlix)

[power-flow-card-plus](https://github.com/flixlix/power-flow-card-plus)

But correct home consumption is always wrong, as power consumption of the inverter is not used for calculation.

so same on my side: load_Power_RST does have constant values not at all same as my house consumption.

just my view cents.

Found this:

![85798e5fdc744fa69f16647d2f97f5a2ff92bd45_2_690x250](https://github.com/nathanmarlor/foxess_modbus/assets/145346227/bfb73386-809b-4882-a2a3-bdaeb44cc6b1)

is load power related to EPS only?

**Reply by vip-klaus** [reply]
Hello,

please excuse my late reply. With the two changes mentioned here in the entity_descriptions.py ('validate=[Range(-100, 100)]' // remove of: post_process=lambda x: max(x, 0)) the values for the inverter power now displayed correctly - including the negative values while charging the battery with additional power from the slave inverter!
The inverter current remains positive, although mathematically correct it should actually be negative. But it's the same with the FoxESS Cloud.
From my point of view, FoxESS-Modbus now works absolutely correctly!

Many thanks to everyone for the competent support!

Best regards
Klaus

![InvPower_FoxESS-Cloud vs  HA-Modbus_01 10 2023](https://github.com/nathanmarlor/foxess_modbus/assets/144471901/6f759a01-1f53-47f8-9f11-3fc39032ed74)

**Reply by canton7** [reply]
Thanks! I'll get this released soon

**Comment by Lich2023** [comment]
I have one more question.
I read This:
https://github.com/nathanmarlor/foxess_modbus/wiki/Supported-Features

Is there still No Support for the H3 to Read/write charge periods and Read/write work mode, min/max SoC or does this now also Work with the H3 too?
I am also very interessted to so This with the H3:
https://github.com/nathanmarlor/foxess_em

Does this Work now for the FoxESS H3 correctly?
What modbus adapters can you prefer for the H3?

**Reply by canton7** [reply]
No, the H3 does not support setting charge periods over modbus unfortunately.

**Reply by Lich2023** [reply]
Is This Not possible or will it be supported in the Future?

**Reply by Lich2023** [reply]
Any Other possibilities to Change anything in HA with the H3? I need This for automatations. :-(

**Reply by Lich2023** [reply]
Force start and Stop for charging over the grid also Not working to Trigger with ha? Do you have any ideas for that?

**Reply by calum-mcfarlane** [reply]
For the features you mention to work, they first have to be supported by the hardware itself. Looking here https://github.com/nathanmarlor/foxess_modbus/wiki/Supported-Features it would very much seem that the H3 doesn't support changing parameters over RS485. I think the modbus registers on the H3 are fairly well understood at this point, so if the hardware doesn't support it at a firmware level, there isn't anything that the developers of the integration can do about it.

I think your best bet would be to contact FoxESS and ask them to add this feature to the H3 range. Of course I can't promise that they will do it, but the only chance to get this working is for Fox to first implement it in the device hardware. **If** that happens, I don't think it would be a huge amount of work for support to be added to the integration for changing parameters.

**Reply by vip-klaus** [reply]
By the way: has anyone heard of the soon-to-be-released Q3 inverter (aka H3-pro) (up to 30 kW, three trackers)?
Will it then also be supported by FoxESS-Modbus? I would like to install one of these in my PV system as quickly as possible...

**Reply by Lich2023** [reply]
Ok thank you...do you know any Other inverters. Who can so This ha intergration of Features better than Fox?

**Reply by calum-mcfarlane** [reply]
There is this project https://github.com/britkat1980/giv_tcp which allows monitoring of GivEnergy inverters. I don't know how well it works with their [three phase gear](https://givenergy.co.uk/products/hybrid-inverter/) but I have the impression that GivEnergy are maybe a bit more joined up in their development compared to Fox. Worth checking with them on their repo anyhow.

(I have no involvement in the giv_tcp project, but my Mum recently got solar and storage and she has their equipment, so I've been having a little poke around to see what's what. Their cloud platform is much better than the Fox one so I haven't had to stand up a HA instance for her!)

**Reply by Lich2023** [reply]
What about sungrow or goodwe...nur i dont know If the Funktion "load battery  from grind " exists

**Reply by Lich2023** [reply]
I want to use evcc.

**Reply by calum-mcfarlane** [reply]
No idea I'm afraid! Probably best to contact the manufacturers first (or at least read up on their products) to find out what support they have for Modbus, MQTT, or whatever - ideally some kind of open standard that HA can use.  Then you're not dependent on a specific integration per se. Searching on Github is a pretty good place to start if you are looking for projects developed for a particular brand of solar kit.

**Reply by TonyM1958** [reply]
You can give this a try if you wish: https://github.com/TonyM1958/FoxESS-Cloud/wiki/Running-in-Google-Colaboratory

This has been tested with H1 but not H3 (I believe the API calls are identical across Fox inverter models).

You don't need Home Assistant, as it uses the Fox ESS Cloud API to read and update your charge settings and, optionally, change work modes. By default, the code won't change any settings when you run it, but it will look at your generation and consumption and the state of your battery and recommend what charge might be needed. You should add Solcast so it has a solar forecast for tomorrow as it will then be a lot more accurate. Due to the limited number of API calls to Solcast, downloading forecast is restricted to certain times by default - but I can tell you how to update that if required and how to enable it to update your charge settings if you wish. 

Here's an example of what it produces when I run it in Jupyter Lab / Colab:

![image](https://github.com/nathanmarlor/foxess_modbus/assets/63789168/604980fb-0f80-4224-8974-9574bbae8db2)

**Reply by TonyM1958** [reply]
One thing worth noting: Google Colab sets local time to GMT but the code adjusts this to BST. If you are in a different time zone, add the parameter time_shift=n to the call to charge_needed. This disables BST and adds / subtracts the number of hours (n) you specify to GMT. Check the time is correct under Parameters when it runs, otherwise, the battery discharge and charge times will be off.

Charge times are configured via f.tariff, by default this is octopus_flux but you can change the tariff or setup custom_periods.

**Comment by benbeton** [comment]
what kind of template sensor do you guys use to monitor the current load of the house, whether the power comes from the grid, the battery or the PV system

**Reply by mafischerhei** [reply]
@benbeton : maybe you could try if this works for you, too?

https://github.com/nathanmarlor/foxess_modbus/discussions/462

**Comment by mafischerhei** [comment]
Houseload

I think the easiest and most accurate results can be get by simply adding: (no matter if we have solar, solar export or battery power)
By just adding 6 sensors together I get very close results as the foxes app gives:
(Maybe we can have this sensor in this integration here as "household" with an update:

Grid Consumption R (foxess)
+ Grid Consumption S (foxess)
+ Grid Consumption T (foxess)
+ Inverter Power R (foxess)
+ Inverter Power S (foxess)
+ Inverter Power T (foxess)

Done

(Would be even better to add all three Inverter ESP outlets, but as I can see we don't have those currently in this integration, do we?


EDIT: I made a little correction of my above calculation. You can have a look here:
https://github.com/nathanmarlor/foxess_modbus/discussions/462

**Reply by canton7** [reply]
They're there, they're just disabled by default so you need to enable them.
